releasenotes/sfdx/README.md

```diff
@@ -21,10 +21,99 @@ Additional documentation:
 
+* NEW:  Quickly gather Salesforce CLI configuration data and run diagnostic tests with the new `doctor` command. Use the command to easily generate informational files that you can attach to [GitHub issues](https://github.com/forcedotcom/cli/issues) or provide to Salesforce Customer Support. 
+
+    Run without parameters, the command first displays basic information, such as whether you're on the latest CLI version. It then writes your configuration and a detailed diagnosis to a JSON file in the current directory. Use the `--outputdir` parameter to specify a different directory. For example:
+
+    `sfdx doctor --outputdir diagnostic-files`
+
+    Use the `--command` parameter to run a specific command in debug mode; the doctor writes both stdout and stderr to separate `*.log` files. Encapsulate the command in double quotes. For example:
+
+    `sfdx doctor --command "force:org:list --all"`
+
+    To run diagnostic tests on a specific plugin rather than the CLI itself, use the `--plugin` parameter. If the plugin isn't listening to the doctor, then you get a warning and no data.
+
+    `sfdx doctor --plugin @salesforce/plugin-source`
+    
+    We've made it really easy to create a GitHub issue: specify the `--createissue` parameter, enter a title at the prompt, and a browser window automatically opens with a partially-filled GitHub issue. Enter the remaining information about your specific issue, click **Submit new issue**, and you're done.  Easy peasy!
+
+    The CLI doctor is in and ready to diagnose all your problems!
+    
+* NEW: You can now automatically replace snippets of your metadata source files with specific values right before you deploy the files to an org with the `force:source:deploy|push` commands. Use this new feature to, for example, replace the endpoint in a NamedCredential, depending on whether you're deploying to a production or scratch org. Or specify a password in an ExternalDataSource that you don't want to store in your repo. The use cases are endless!
+
+    To configure string replacement, add a `replacements` property to your `sfdx-project.json` file and use key-value pairs to describe how the string replacement works. 
+
+    For example, this `sfdx-project.json` snippet specifies that when you deploy the `force-app/main/default/classes/myClass.cls` source file, all occurrences of the string `replaceMe` are replaced with the value of the `THE_REPLACEMENT` environment variable:
+
+    ```bash
+    {
+      "packageDirectories": [
+         {
+           "path": "force-app",
+           "default": true
+         }
+      ],
+      "name": "myproj",
+      "replacements": [
+        {
+          "filename": "force-app/main/default/classes/myClass.cls",
+          "stringToReplace": "replaceMe",
+          "replaceWithEnv": "THE_REPLACEMENT"  
+        }
+      ],
+    …
+    }
+    ```
+
+    You can specify these keys in the `replacements` property:
```

Comment:
I *think* we want to point out the 2 places people are likely to run into problems
1. always use forward slashes, even on windows
2. JSON requires that you escape any backslashes with another backslash

Reply:
Added